src/v3/src/types/schema.ts

```diff
@@ -58,33 +58,33 @@ export type WidgetMessage = Modify<IdxMessage, {
   title?: string;
   name?: string;
   description?: string;
-  links?: WidgetMessageLink[];
+  links?: MessageLink[];
+  options?: WidgetMessageOption[];
   listStyleType?: ListStyleType,
 }>;
 
 export type ListStyleType = 'circle' | 'disc' | 'square' | 'decimal';
 
+export type WidgetMessageOption = WidgetMessageLinkButton | WidgetMessageLink | WidgetMessageText;
+
 export type MessageLink = {
   label: string;
   url: string;
 };
 
-/**
- * When URL is missing from object, link will be treated as a button link
- * utilizing the onClick handler
- */
-export type WidgetMessageLink<TIsLinkButton = boolean> = TIsLinkButton extends true
-  ? {
-    isLinkButton: TIsLinkButton;
-    label: MessageLink['label'];
-    onClick: () => void;
-    dataSe: string;
-  }
-  : {
-    isLinkButton: TIsLinkButton;
-    label: MessageLink['label'];
-    url: MessageLink['url'];
-  };
+export type WidgetMessageLinkButton = {
```

Comment:
Can we consider a non-exported base type that lists the common properties and have the others extend it?